File: backend/app/core/categories.py
CATEGORIES = [
    "essentials",
    "grocery",
    "maintainance",
    "misc",
    "recharge",
    "rent/repay",
    "shopping",
    "utilities",
    "cash",
]
KEYWORD_MAP = {
    "rent": "rent/repay",
    "emi": "rent/repay",
    "loan": "rent/repay",
    "admission": "rent/repay",
    "fee": "rent/repay",
    "lpg": "utilities",
    "electricity": "utilities",
    "wifi": "utilities",
    "broadband": "utilities",
    "gas": "utilities",
    "water bill": "utilities",
    "recharge": "recharge",
    "mobile": "recharge",
    "prepaid": "recharge",
    "medical": "essentials",
    "medicine": "essentials",
    "pharmacy": "essentials",
    "hospital": "essentials",
    "doctor": "essentials",
    "grocery": "grocery",
    "groceries": "grocery",
    "mart": "grocery",
    "vegetable": "grocery",
    "milk": "grocery",
    "dudh": "grocery",
    "repair": "maintainance",
    "maintenance": "maintainance",
    "maintainance": "maintainance",
    "service": "maintainance",
    "flipkart": "shopping",
    "amazon": "shopping",
    "myntra": "shopping",
    "shopping": "shopping",
    "cash": "cash",
    "atm": "cash",
    "withdrawal": "cash",
}


def guess_category(text: str) -> str:
    text = (text or "").lower()
    for keyword, category in KEYWORD_MAP.items():
        if keyword in text:
            return category
    return "misc"

File: frontend/src/constants/categories.js

export const CATEGORIES = [
  { id: 'essentials', name: 'Essentials', icon: '🧾' },
  { id: 'grocery', name: 'Grocery', icon: '🛒' },
  { id: 'maintainance', name: 'Maintainance', icon: '🔧' },
  { id: 'misc', name: 'Misc', icon: '✨' },
  { id: 'recharge', name: 'Recharge', icon: '📱' },
  { id: 'rent/repay', name: 'Rent/Repay', icon: '🏠' },
  { id: 'shopping', name: 'Shopping', icon: '🛍️' },
  { id: 'utilities', name: 'Utilities', icon: '💡' },
  { id: 'cash', name: 'Cash', icon: '💵' },
];

export const categoryIcon = (category) => {
  const match = CATEGORIES.find(
    (c) => c.id === String(category || '').toLowerCase()
  );
  return match ? match.icon : '✨';
};